tests/pretty/delegation-inherit-attributes.rs

```diff
@@ -52,5 +52,49 @@ reuse to_reuse_functions::const_fn_extern;
 #[must_use = "some reason"]
 reuse to_reuse_functions::async_fn_extern;
 
+mod recursive {
+    // Check that `baz` inherit attribute from `foo`
+    mod first {
+        fn bar() {}
+        #[must_use = "some reason"]
+        reuse bar as foo;
+        reuse foo as baz;
+    }
+
+    // Check that `baz` don't inherit attribute from `bar`
+    mod second {
+        #[must_use = "some reason"]
+        fn bar() {}
+
+        reuse bar as foo;
+        reuse foo as baz;
+    }
+
+    // Check that `foo5` don't inherit attribute from `bar`
+    // and inherit attribute from foo4
+    mod third {
+        #[must_use = "some reason"]
+        fn bar() {}
+        reuse bar as foo1;
+        reuse foo1 as foo2;
+        reuse foo2 as foo3;
+        #[must_use = "foo4"]
+        reuse foo3 as foo4;
+        reuse foo4 as foo5;
```

Comment:
Similarly, I'd expect `foo5` and `foo4` to have `must_use = "foo4"`, and `foo3`, `foo2`, `foo1` and `bar` to have `must_use = "some reason"`.